controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -94,58 +101,22 @@ public SegmentHelper(final ConnectionFactory clientCF, HostControllerStore hostS
         final String qualifiedStreamSegmentName = getQualifiedStreamSegmentName(scope, stream, segmentId);
         final Controller.NodeUri uri = getSegmentUri(scope, stream, segmentId);
-        final WireCommandType type = WireCommandType.CREATE_SEGMENT;
-        final long requestId = (clientRequestId == RequestTag.NON_EXISTENT_ID) ? idGenerator.get() : clientRequestId;
-
-        final FailingReplyProcessor replyProcessor = new FailingReplyProcessor() {
-            @Override
-            public void connectionDropped() {
-                log.warn(requestId, "CreateSegment {} Connection dropped", qualifiedStreamSegmentName);
-                result.completeExceptionally(
-                        new WireCommandFailedException(type, WireCommandFailedException.Reason.ConnectionDropped));
-            }
-
-            @Override
-            public void wrongHost(WireCommands.WrongHost wrongHost) {
-                log.warn(requestId, "CreateSegment {} wrong host", qualifiedStreamSegmentName);
-                result.completeExceptionally(new WireCommandFailedException(type, WireCommandFailedException.Reason.UnknownHost));
-            }
-
-            @Override
-            public void segmentAlreadyExists(WireCommands.SegmentAlreadyExists segmentAlreadyExists) {
-                log.info(requestId, "CreateSegment {} segmentAlreadyExists", qualifiedStreamSegmentName);
-                result.complete(true);
-            }
-
-            @Override
-            public void segmentCreated(WireCommands.SegmentCreated segmentCreated) {
-                log.info(requestId, "CreateSegment {} SegmentCreated", qualifiedStreamSegmentName);
-                result.complete(true);
-            }
-
-            @Override
-            public void processingFailure(Exception error) {
-                log.error(requestId, "CreateSegment {} threw exception", qualifiedStreamSegmentName, error);
-                handleError(error, result, type);
-            }
-
-            @Override
-            public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenCheckFailed) {
-                result.completeExceptionally(
-                        new WireCommandFailedException(new AuthenticationException(authTokenCheckFailed.toString()),
-                                type, WireCommandFailedException.Reason.AuthFailed));
-            }
-        };
 
+        RawClient connection = new RawClient(ModelHelper.encode(uri), connectionFactory);
+        final long requestId = connection.getFlow().asLong();
         Pair<Byte, Integer> extracted = extractFromPolicy(policy);
 
-        WireCommands.CreateSegment request = new WireCommands.CreateSegment(requestId, qualifiedStreamSegmentName,
-                extracted.getLeft(), extracted.getRight(), controllerToken);
-        sendRequestAsync(request, replyProcessor, result, ModelHelper.encode(uri));
-        return result;
+        Map<Class, Runnable> expectedReplies = new HashMap<>();
```

Comment:
You are repeating the same pattern over and over again and creating a new Map just to put two elements in each. This makes the code look bad and is inefficient.

I recommend you modify your `transformReply` method to extract the necessary info (`Class.getSimpleName()` will get you the name of the class, and you can pass in a String with the operation name) and do a more generic error handler.

Reply:
Thanks for the comments Andrei. this has been fixed